README.md

```diff
@@ -1,205 +1,278 @@
+
+#### Code Development <a name="rdevel"/>
+
+Code development in our R repositories is complex. To navigate this complexity we use a combination of software tools ([RStudio](https://www.rstudio.com/products/rstudio/download/) and [Atom]()) and testing. This complexity is rooted in our software architecture where many times changes to R code has an effect on another R package and/or the C code. Our R packages revolve around increasing user functionality of our SOILWAT2 C model, so even if code changes only occur in R, you must be conscientious of potential downstream effects.
+
+For example, changes in rSOILWAT2 will almost always require changes to SOILWAT2 (i.e. so that SOILWAT2 can receive a new variable that is being passed from rSOILWAT2) and rSFSW2 (i.e. rSFSW2 would format site-specific information and then use a rSOILWAT2 function to format this information for passage to C).
+
+Here are some guidelines to assist in R code development:
+
+* If your code is a feature, open up a new project on GitHub and sketch out a plan (specific functions) for how you will implement new functionality. Ask for review from your supervisor.
+* Take advantage of [RStudio's debugger](https://support.rstudio.com/hc/en-us/articles/205612627-Debugging-with-RStudio) to walk through functions with similar functionality.
+  - Note: Ironically, sometimes the debugger can be buggy. With patience, you can still walk through functions.
+* Test your new code within an integration project.  
+  - Copy the test project from within the _/Git/_ folder to your _/Desktop_ so that changes to input files aren't integrated onto Git.
+  - Ensure you can run the test project in RStudio prior to any code development. See User Manual (todo) for running a test project.
+  - Change or set options in the descriptions.R file so that your new functionality is triggered (pertains to rSFSW2).
+  - Switching on and off of certain options should NOT be added into _GIT_. If a _new_ option is added though, this should be permanently integrated into the code base (i.e. for rSFSW2 in the _/demo_ folder and the _test project_ folder).
+* Develop code in Atom. To test code, re-install the R package and then re-run the test project within RStudio.
+* Document your code both formally (see [R documentation](#rdoc)) and informally (in-line with code explaining reasoning using #).
+* Run the test project again with your new functionality. Because of the labyrinthian nature of the code base you may have missed something you need to change and will receive an error message. Decipher and fix error message, re-install, and try again. shift + Ctrl + F is your friend.
+* Ensure your new code adheres to our style guide. Take advantage of lintr.
+* Ensure your package passes all pre-existing unit tests.
```

Comment:
I would add `locally` -- as distinct from `CIs`

Reply:
I would also add that it is possible to run individual unit tests and/or individual unit test files -- in case, there is just one/few problems and it doesn't make sense (waste of time) to run the entire set.